### PR TITLE
Fix duration check

### LINKED ISSUE #53: Confusing behavior around duration config loaded from external file
### Steps to reproduce
1. Create an _index.js_ file and paste the following code:
    ```js
var convict = require('convict');

var conf = convict({
      session: {
        doc: 'a doc',
        format: 'duration',
        default: '2 days'
      }
});

conf.loadFile('./config.json');
conf.validate();

console.log(conf.get('session'));
```

2. Create a _config.json_ file in the same directory as _index.js_ created in step 1:
    ```json
{
      "session": "4 days"
}
```

3. Run `node index`

### Actual results:
```sh
$ node index

/Users/pdehaan/dev/tmp/convict-test/node_modules/convict/lib/convict.js:393
        throw new Error(errBuf);
              ^
Error: session: must be a positive integer: value was "2 days"
    at Object.rv.validate (/Users/pdehaan/dev/tmp/convict-test/node_modules/convict/lib/convict.js:393:15)
    at Object.<anonymous> (/Users/pdehaan/dev/tmp/convict-test/index.js:12:6)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:906:3
```

This seems to be because of the "2 days" in the _config.json_ file. It works fine when I set the "default" value in the _index.js_ but implodes horribly in the JSON file.

### Workaround:
Set the duration as milliseconds in the `config.json` file.

### PR DESCRIPTION
**Pull request by madarche**
This PR fixes #53.

### CODE REVIEW

**Comment by madarche** [comment]
For the record, the duration check in human readable string was (unintentionally I believe) removed with commit https://github.com/mozilla/node-convict/commit/b7ace0540789cacc3af38fce485d51c69521c74f#diff-7763d1369911b91b757ba7ff51d734a2L37. I say unintentionally because the `coerce` function has been supporting the duration in human readable string until now.

@zaach could you review this PR please? Thank you.

**Comment by zaach** [comment]
Ah, good catch. :+1:

File: README.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ convict provides serveral predefined formats for validation that you can use ([u
 * `url`
 * `email`
 * `ipaddress` - IPv4 and IPv6 addresses
-* `duration` - miliseconds or a human readable string (e.g. "5 days")
+* `duration` - milliseconds or a human readable string (e.g. "3000", "5 days")
```
Comment (6a68) [inline README.md]: @zaach @madarche should this actually say `(e.g. 3000, "5 days")`? This example gives milliseconds as a string :-\

File: lib/convict.js [hunk 2]
```diff
@@ -50,7 +50,12 @@ var types = {
     assert(validator.isIP(x), 'must be an IP address');
   }, "There is no IPv4 vs IPv6 addresses checking anymore, use \"ipaddress\" instead."),
   duration: function(x) {
-    assert(validator.isInt(x) && x >= 0, 'must be a positive integer');
+    var err_msg = 'must be a positive integer or human readable string (e.g. "3000", "5 days")';
```
Comment (6a68) [inline lib/convict.js]: ditto here

File: test/cases/file_basic.json [hunk 3]
```diff
@@ -1,4 +1,6 @@
 {
   "ip": "10.0.1.101",
-  "port": 8080
+  "port": 8080,
+  "session": "4 days",
+  "cache": "3000"
```
Comment (6a68) [inline test/cases/file_basic.json]: again, do we want to be enclosing integers in quotes?

File: test/cases/file_basic.out [hunk 4]
```diff
@@ -1,4 +1,6 @@
 {
   "ip": "10.0.1.101",
-  "port": 8080
-}
\ No newline at end of file
+  "port": 8080,
```
Comment (6a68) [inline test/cases/file_basic.out]: we aren't enclosing the port value in quotes, idk

**Comment by 6a68** [comment]
Thanks for taking care of this, @madarche and @zaach! I'm excited to replace `604800000` with `"1 week"` in my configs ^_^

File: README.md (hunk 1 above)
Comment (zaach) [inline README.md]: Hrm, that should not be a string.

Comment (madarche) [inline README.md]: @6a68 @zaach I'm writing a fix.

**Pull request merged**